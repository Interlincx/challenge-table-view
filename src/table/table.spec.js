import React from 'react'
import Table from './table'
import { TableBody, TableCell, TableHead, TableRow } from '@material-ui/core'

const minProps = {
  columns: [
    {
      label: 'Name',
      dataKey: 'name'
    },
    {
      label: 'Age',
      dataKey: 'age'
    }
  ],
  rows: [
    { name: 'Mike', age: 25, id: 1245 },
    { name: 'Jennifer', age: 30, id: 46874 }
  ]
}

test('renders a Table', () => {
  const wrapper = shallow(<Table {...minProps} />)
  expect(wrapper).toMatchSnapshot()
})

test('should have 3 rows', () => {
  const wrapper = mount(<Table {...minProps} />)
  expect(wrapper.find(TableRow)).toHaveLength(3)
})

test('should have 1 table head', () => {
  const wrapper = mount(<Table {...minProps} />)
  expect(wrapper.find(TableHead)).toHaveLength(1)
})

test('should have 1 table body', () => {
  const wrapper = mount(<Table {...minProps} />)
  expect(wrapper.find(TableBody)).toHaveLength(1)
})

test('should have 6 cells', () => {
  const wrapper = mount(<Table {...minProps} />)
  expect(wrapper.find(TableCell)).toHaveLength(6)
})
