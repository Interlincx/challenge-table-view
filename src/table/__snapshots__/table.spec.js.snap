// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Table 1`] = `
<Table
  classes={
    Object {
      "root": "Table-root-1",
      "table": "Table-table-2",
    }
  }
  columns={
    Array [
      Object {
        "dataKey": "name",
        "label": "Name",
      },
      Object {
        "dataKey": "age",
        "label": "Age",
      },
    ]
  }
  rows={
    Array [
      Object {
        "age": 25,
        "id": 1245,
        "name": "Mike",
      },
      Object {
        "age": 30,
        "id": 46874,
        "name": "Jennifer",
      },
    ]
  }
/>
`;
